driver/src/main/java/org/neo4j/driver/v1/Config.java

```diff
@@ -117,21 +129,26 @@ public boolean logLeakedSessions()
 
     /**
      * Max number of connections per URL for this driver.
+     *
      * @return the max number of connections
+     * @deprecated please use {@link #maxConnectionPoolSize()} instead.
      */
     @Deprecated
     public int connectionPoolSize()
     {
-        return maxIdleConnectionPoolSize;
+        return maxConnectionPoolSize;
     }
 
     /**
      * Max number of idle connections per URL for this driver.
+     *
      * @return the max number of connections
+     * @deprecated please use {@link #maxConnectionPoolSize()} instead.
```

Comment:
I would suggest changing to `Please consider to use #maxConnectionPoolSize` and `#MaxConnectionLifetime` to replace this method instead.

Same applies for comments of all `MaxIdle` methods

Reply:
Improved javadoc with your suggestion for `withMaxSessions`, `withMaxIdleSessions`, and `withMaxIdleConnections`.